### PR TITLE
Initial implementation of initiative

### PR DESCRIPTION
**Pull request by andrewrula**

### CODE REVIEW

File: combatEngine [hunk 1]
```diff
@@ -60,22 +61,51 @@ function addFighter(type,team){
...
 function haveBattle(){
     var turncount = 0
     while (fighters[0].hp > 0 && fighters[1].hp > 0){
         turncount = turncount + 1;
         console.log("Begin Turn " + turncount + ".")
         //Establish Initiative Order
         var initiativeOrder = getInitiative();
+        
+        
+        //Ends battle immediately
```
Comment (andrewrula) [inline combatEngine:101]: Stops infinite loop

File: combatEngine [hunk 2]
```diff
@@ -60,22 +61,51 @@ function addFighter(type,team){
...
     var needInitiative = fighters
-    var iterate = 0    
-    while (needInitiative.length > 0){}
+    //Finds the highest initiative of the unassigned units and moves it to the initiativeOrder array, repeats until no more need Initiative, then returns that array. Ties go to the unit earlier in the list.
+    while(needInitiative.length > 0){
+        if(wantLogs = true){
+            console.log("need Initiative is length " + needInitiative.length + " and contains: ")
+            console.log (needInitiative)
+        };
+        topInit = -999
```
Comment (Kolbeckenstein) [inline combatEngine:72]: If using strat, use https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/MIN_SAFE_INTEGER instead.

File: combatEngine [hunk 3]
```diff
@@ -60,22 +61,51 @@ function addFighter(type,team){
...
+            if(wantLogs == true){
+                console.log("i is " + i);
+                console.log("current i's speed is: " + needInitiative[i].speed);
+            };
+            if (needInitiative[i].speed > topInit){
+                topInit = fighters[i].speed
+                topInitIndex = i
+            };
+        };
+        initiativeOrder.push(needInitiative.splice(topInitIndex,1))
```
Comment (Kolbeckenstein) [inline combatEngine:83]: Long term replace this with a sort, but maybe only after you have a more complex combat object.

File: combatEngine [hunk 4]
```diff
@@ -60,22 +61,51 @@ function addFighter(type,team){
...
+        console.log("Initiative Order Is:")
+    console.log(initiativeOrder);
+    };
+    return initiativeOrder; 
 };
-        
+
 function haveBattle(){
     var turncount = 0
     while (fighters[0].hp > 0 && fighters[1].hp > 0){
```
Comment (Kolbeckenstein) [inline combatEngine:94]: Avoid hard codes like this!
maybe something like...    while(fighters.filter(x => x.hp > 0).length > 0){ ...

**Review by Kolbeckenstein — APPROVED** [review]

File: combatEngine [hunk 5]
```diff
@@ -60,22 +61,51 @@ function addFighter(type,team){
 };
 
 function getInitiative(){//Helper Function to create an initiative order.
+    var initiativeOrder = [];
     var needInitiative = fighters
```
Comment (Kolbeckenstein) [inline combatEngine:65]: So this is assigning to the variable needInitiative a reference to the same array that fighters is pointing to.

If you want a new array, use `needInitiative = [...fighters]`

**Review by Kolbeckenstein — APPROVED** [review]

**Review by Kolbeckenstein — APPROVED** [review]

File: combatEngine (hunk 3 above)
Comment (andrewrula) [reply combatEngine:83]: Yea, I figure that the vast majority of the infrastructure here is doomed to start so this is just a quick way to count to like....20-30 combat objects with initiative and get a rough sort in a way that doesn't break

File: combatEngine (hunk 4 above)
Comment (andrewrula) [reply combatEngine:94]: TIL the filter method. Hold please.

Comment (andrewrula) [reply combatEngine:94]: Also off by 1, otherwise looks good.  We want to fight until there's 1 left, not nothing left.

**Review by Kolbeckenstein — APPROVED** [review]

**Pull request merged**